runtime/queries/metricsview.go

```diff
@@ -297,7 +297,7 @@ func columnIdentifierExpression(mv *runtimev1.MetricsViewSpec, aliases []*runtim
 	// check if identifier is a dimension
 	for _, dim := range mv.Dimensions {
 		if dim.Name == name {
-			return safeName(metricsViewDimensionColumn(dim)), true
+			return metricsViewDimensionExpression(dim), true
```

Comment:
This will get templated into the where clause:

1. Might it be problematic to put a free form expression in here? I think at the least it will need to be wrapped in parentheses
2. Would it be better to template in just the dimension name (which the expression is aliased to in the `SELECT` clause)? But not sure if that works on Druid

Reply:
The problem with this is for queries where the dimension is not selected we would have issues with the column not being there. The other option is to only add it if is in the select caluse. That made these builder functions too complex.

As for adding parentheses, it gets added when the identifier is actually used.

I guess we are missing a validation to make sure there isnt sql injection happening.

Reply:
1. The case where the dim is not selected makes sense. Agree on not making it conditional on the select clause, seems to complicated.
2. About the parentheses, imagine if the expression is `col1 = 10` and the filter is `dim1 = true`, then you're saying the parenthesis will be added such that the where clause becomes `(col1 = 10) = true` (correct) and not `col1 = 10 = true` (which would error)?
3. How could SQL injection happen? For the where filters, I thought we were doing lookups to resolve identifiers and using `?` and args to pass values?

Reply:
2. Yes all binary expressions have it. `like` and `in` do not seem to have it. Lemme add that.
3. It could happen in the dimension expression, `count(*)";drop table blah; --` or something like that.